### PR TITLE
ESM2 attention_mask and token_dropout fix

### PR DESCRIPTION
**Pull request by lhallee**
Fixes:
https://github.com/huggingface/transformers/issues/44162

@ArthurZucker @Cyrilvallez

### CODE REVIEW

**Review by zucchini-nlp — APPROVED** [review]
Thanks, lgtm! Can you check why the `embeds` test is failing, I think we might have to skip it or disable `token_dropout` for this particular test

**Comment by Rocketknight1** [comment]
I think that test is invalid when embeddings are scaled because of token dropout, so I'll add a skip so we can merge this quickly

**Comment by Rocketknight1** [comment]
run-slow: esm
